modules/core/src/main/java/org/apache/synapse/rest/cors/SynapseCORSConfiguration.java

```diff
@@ -42,7 +50,30 @@ private SynapseCORSConfiguration() {
         //Retrieve allowed origin list
         String allowedOriginListStr =
                 SynapsePropertiesLoader.getPropertyValue(RESTConstants.CORS_CONFIGURATION_ACCESS_CTL_ALLOW_ORIGIN, "");
-        allowedOrigins = new HashSet<>(Arrays.asList(allowedOriginListStr.split(",")));
+        String[] originList = allowedOriginListStr.split(",");
+        for (String origin : originList) {
+            String trimmedOrigin = origin.trim();
+            allowedOrigins.add(trimmedOrigin);
+            try {
+                URL url = new URL(trimmedOrigin);
+                if (url.getHost().equals("localhost")) {
+                    // Add localhost IPs as allowed origin
+                    Enumeration<NetworkInterface> networkInterfaces = NetworkInterface.getNetworkInterfaces();
+                    while (networkInterfaces.hasMoreElements()) {
+                        NetworkInterface nInterface = networkInterfaces.nextElement();
+                        for (InterfaceAddress iAddr : nInterface.getInterfaceAddresses()) {
+                            URL localUrl =
+                                    new URL(url.getProtocol(), iAddr.getAddress().getHostAddress(), url.getPort(), "");
+                            allowedOrigins.add(localUrl.toString());
+                        }
+                    }
+                }
+            } catch (MalformedURLException e) {
+                LOG.error("Provided origin URL "+ trimmedOrigin + " is malformed", e);
```

Comment:
only logging is enough here?

Reply:
Fixed, I was under impression upstream code won't handle the exception. With further investigation found it will execute the fault sequence. So we can throw instead of logging